frontend/apps/reader/modules/readerview.lua

```diff
@@ -724,13 +724,21 @@ end
 
 function ReaderView:onReadSettings(config)
     self.render_mode = config:readSetting("render_mode") or 0
-    local locked = G_reader_settings:readSetting("lock_rotation")
-    local rotation_mode = config:readSetting("rotation_mode")
-    if not rotation_mode and locked then
-        if self.ui.document.info.has_pages then
-            rotation_mode = G_reader_settings:readSetting("kopt_rotation_mode") or 0
-        else
-            rotation_mode = G_reader_settings:readSetting("copt_rotation_mode") or 0
+    local rotation_mode = nil
+    local locked = G_reader_settings:isTrue("lock_rotation")
+    if locked then
+        -- Keep FM rotation
+        rotation_mode = G_reader_settings:readSetting("fm_rotation_mode") or Screen.ORIENTATION_PORTRAIT
```

Comment:
This will reset the rotation to the default FM rotation not the current FM rotation as is desired.

it should be:
``` lua
local locked = G_reader_settings:isTrue("lock_rotation")
if  not locked then
        -- Honor docsettings's rotation
        rotation_mode = config:readSetting("rotation_mode") -- Doc's
        if not rotation_mode then
```

The current rotation will be kept by doing nothing

Reply:
Oh, yeah, I was approaching that slightly differently, but that actually makes much more sense ;).

And points towards making `lock_rotation` actually do what you probably originally meant (as is, kepe the current rotation *both* ways when going FM <-> reader).